pandas/io/formats/html.py

```diff
@@ -85,8 +85,11 @@ def write(self, s, indent=0):
         rs = pprint_thing(s)
         self.elements.append(' ' * indent + rs)
 
-    def write_th(self, s, indent=0, tags=None):
-        if self.fmt.col_space is not None and self.fmt.col_space > 0:
+    def write_th(self, s, header=False, indent=0, tags=None):
+        if header and self.fmt.col_space is not None:
+            if isinstance(self.fmt.col_space, int):
+                self.fmt.col_space = ('{colspace}px'
```

Comment:
Might be wrong but wouldn't this always append px, even if the user specified another unit?

Reply:
nah, it only does that if the user passed in an `int`.  If they specified units (by passing a string) then it does nothing.

Reply:
Ah sorry missed the isinstance check. Thanks for clarifying